Unity-Technologies/Graphics
# Fixes for 3 cases of undo not working

*Summary:*
Fixes bugs related to lack of registering undos: 1 prio 3 bug and 3 other Undo related bugs (first two are basically the same bug):
- https://fogbugz.unity3d.com/f/cases/1184932/
- https://fogbugz.unity3d.com/f/cases/1215447/
- https://fogbugz.unity3d.com/f/cases/1215448/
- https://fogbugz.unity3d.com/f/cases/1213629/

[Internal link to previous PR](https://github.com/Unity-Technologies/ScriptableRenderPipeline/pull/6357)

*Manually Tested:*

That none of the bug reproduces anymore
For the default-value-sub-node, all sorts of changes to confirm that the title asterisk only appears when necessary, such as something that results in the field returning to it was before type (0, safeaef => 0)

Technical Risk: 1/4
Halo Effect: 1/4
